controller/formation.go

```diff
@@ -303,7 +318,14 @@ func (r *FormationRepo) unsubscribeLocked(ch chan *ct.ExpandedFormation) {
 		for range ch {
 		}
 	}()
+	si, ok := r.subscriptions[ch]
 	delete(r.subscriptions, ch)
+	if ok {
+		si.uc <- struct{}{}
+		si.wg.Wait()
```

Comment:
We want the `sendUpdatedSince` goroutine to stop if the channel is closed, which we can do by just wrapping the channel:

```go
type FormationSubscription struct {
	mtx sync.Mutex
	ch  chan *ct.ExpandedFormation
}

func (f *FormationSubscription) Notify(ef *ct.ExpandedFormation) bool {
	f.mtx.RLock()
	defer f.mtx.RUnlock()
	if f.ch == nil {
		return false
	}
	f.ch <- ef
	return true
}

func (f *FormationSubscription) Close() {
	f.mtx.Lock()
	defer f.mtx.Unlock()
	close(f.ch)
	f.ch = nil
}

// in sendUpdatedSince
for rows.Next() {
	// ...
	if ok := f.Notify(ef); !ok {
		return nil
	}
}
```

WDYT?